stan/math/prim/arr/functor/mpi_distributed_apply.hpp

```diff
@@ -0,0 +1,49 @@
+#ifdef STAN_MPI
+
+#ifndef STAN_MATH_PRIM_ARR_FUNCTOR_MPI_DISTRIBUTED_APPLY_HPP
+#define STAN_MATH_PRIM_ARR_FUNCTOR_MPI_DISTRIBUTED_APPLY_HPP
+
+#include <stan/math/prim/arr/functor/mpi_command.hpp>
+
+#include <boost/serialization/nvp.hpp>
+#include <boost/serialization/serialization.hpp>
+#include <boost/serialization/access.hpp>
+#include <boost/serialization/base_object.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * MPI command template which calls the static method
+ * distributed_apply of the given class F.
+ *
+ * Distributed MPI commands should be registered using the macro
+ * STAN_REGISTER_MPI_DISTRIBUTED_APPLY called in the root
+ * namespace.
+ *
+ * @tparam F type of functor with static distributed_apply method.
+ */
+template <typename F>
+struct mpi_distributed_apply : public mpi_command {
+  friend class boost::serialization::access;
+  template <class Archive>
+  void serialize(Archive& ar, const unsigned int version) {
+    ar& BOOST_SERIALIZATION_BASE_OBJECT_NVP(mpi_command);
```

Comment:
Please doc that the ampersand here is an operator that's overridden.  Otherwise, it's hard to read the syntax!  Also, space-separate it to make that clearer from looking at it.  (This is more a doc clarity than spacing/formatting thing, I hope.)

Reply:
Same here. This is standard boost serialization. I will add where this mystic notation comes from.

Reply:
BTW, the clang autoformatter "decides" that there may not be a space between ar and &. I did put a space there, but the clang autoformat chew it.

Reply:
OK, I guess the formatter does everything.  Can't fight that as it'd have to know overloads to know how to format `&`.